lib/service-stream.js

```diff
@@ -0,0 +1,54 @@
+
+'use strict';
+
+module.exports = ServiceStream;
+
+function ServiceStream(service, id) {
```

Comment:
I feel like it might make sense to pass the client's `channel` into the `ServiceStream` so that we can pipe messages down it. Right now it looks like we're broadcasting stream messages to all clients. This might mean we can kill the new `._broadcast` API which is confusing me.